src/formBuilder/utils.js

```diff
@@ -691,11 +691,18 @@ function generateSchemaElementFromElement(element: ElementProps) {
       element.schema !== undefined && element.schema.description !== undefined
         ? element.schema.description
         : element.dataOptions.description;
-    return {
+
+    const returnElement = {
       $ref: element.$ref,
       title: title,
       description: description,
     };
+
+    return element.schema !== undefined &&
+      element.schema.required !== undefined &&
+      element.schema.required.length !== 0
+      ? { ...returnElement, required: element.schema.required }
+      : returnElement;
```

Comment:
This is a complex piece of logic to have in a return statement.  It would be better to refactor as:

```js
let returnElement = { ... }

if (condition) {
  returnElement = { ...returnElement, required: element.schema.required }
}

return returnElement;
```

For `(condition)` look if you can rather use the `?.` operator.  Something like this:

```js
const length = element?.schema?.required?.length
if (length !== undefined && length > 0) {
  // ...
}
```